### ISSUE TITLE
Version check in WindowsNetworks is wrong

### ISSUE THREAD

**Issue by Davis8988** [oshi/oshi#431]
Hello,  
Your library is awsome and I am using it for collecting details in my lab.  
In the network speed of my adapters I get a value of **194.7 Mbps** , and when I print the net.getSpeed() method I get the value: 194693632.  
But when I check my adapter properties I can see that it has speed of **1.0 Gbps**  

I am using a switch which supports 1.0 Gbps speed.  Is this an error? or is it really the actual speed?  

Thank you

**Comment by dbwiddis** [comment]
Which operating system are you using?

**Comment by Davis8988** [comment]
Windows 7

**Comment by Davis8988** [comment]
Packets info:
On oshi's output: sent - 1901176  ,  received - 1823106  
On adapter properties in windows: sent - 657,200,157  ,  received - 579,025,716  
  
I can also see in oshi's output that it recognizes the operating system as Window 7.  
  
I included the output and a capture of the adapter properties.  
(where it has **** signs in the output is what I concealed , and other signs like + and # are my prints) 

![capture](https://user-images.githubusercontent.com/26057321/33803445-17bb18a0-dd99-11e7-9ea2-5fd4a260e94b.PNG)
[OSHI_Output.txt](https://github.com/oshi/oshi/files/1545444/OSHI_Output.txt)

**OSHI's Output**:
[main] INFO oshi.SystemInfoTest - Initializing System...
[main] INFO oshi.SystemInfoTest - Checking computer system...
Microsoft Windows 7 build 7600
[main] INFO oshi.SystemInfoTest - Checking Processor...
manufacturer: Hewlett-Packard
model: HP Z210 Workstation
serialnumber: *********
firmware:
  manufacturer: Hewlett-Packard
  name: Default System BIOS
  description: Default System BIOS
  version: HPQOEM - 1072009
baseboard:
  manufacturer: Hewlett-Packard
  model: unknown
  version: unknown
  serialnumber: ********
Intel(R) Xeon(R) CPU E31240 @ 3.30GHz
 4 physical CPU(s)
[main] INFO oshi.SystemInfoTest - Checking Memory...
 4 logical CPU(s)
Identifier: Intel64 Family 6 Model 42 Stepping 7
ProcessorID: ***********
Memory: 1.9 GiB/4.0 GiB
Swap used: 319.5 MiB/4.0 GiB
[main] INFO oshi.SystemInfoTest - Checking CPU...
Uptime: 3 days, 01:14:14
CPU, IOWait, and IRQ ticks @ 0 sec:[30443142, 0, 20874527, 1002740076, 0, 176483, 380517, 0]
CPU, IOWait, and IRQ ticks @ 1 sec:[30443329, 0, 20874698, 1002743773, 0, 176483, 380517, 0]
User: 4.6% Nice: 0.0% System: 4.2% Idle: 91.2% IOwait: 0.0% IRQ: 0.0% SoftIRQ: 0.0% Steal: 0.0%
CPU load: 12.3% (counting ticks)
CPU load: 23.0% (OS MXBean)
CPU load averages: N/A N/A N/A
[main] INFO oshi.SystemInfoTest - Checking Processes...
CPU load per processor: 15.0% 15.1% 12.4% 16.4%
Processes: 63, Threads: 1074
   PID  %CPU %MEM       VSZ       RSS Name
 12064  29.7  1.4   2.4 GiB  57.1 MiB javaw.exe
  1604  11.6  0.8 204.6 MiB  33.0 MiB PRTG Probe.exe
 13196   4.2  0.2  51.3 MiB   8.6 MiB WmiPrvSE.exe
  4988   2.5  0.2  58.0 MiB   8.4 MiB WmiPrvSE.exe
  1012   1.1  0.9 405.6 MiB  36.8 MiB svchost.exe
[main] INFO oshi.SystemInfoTest - Checking Sensors...
Sensors:
[main] ERROR oshi.util.platform.windows.WmiUtil - Could not connect to namespace root\OpenHardwareMonitor. Error code = 0x8004100e
 CPU Temperature: 0.0°C
[main] ERROR oshi.util.platform.windows.WmiUtil - Could not connect to namespace root\OpenHardwareMonitor. Error code = 0x8004100e
 Fan Speeds: [0]
[main] ERROR oshi.util.platform.windows.WmiUtil - Could not connect to namespace root\OpenHardwareMonitor. Error code = 0x8004100e
 CPU Voltage: 0.0V
[main] INFO oshi.SystemInfoTest - Checking Power sources...
[main] INFO oshi.SystemInfoTest - Checking Disks...
Power: Calculating time remaining
 Unknown @ 0.0%
Disks:
 \\.\PHYSICALDRIVE0: (model: ST3250312AS ATA Device (Standard disk drives) - S/N: V9EY9R22            ) size: 250.1 GB, reads: 279706 (5.1 GiB), writes: 806515 (10.0 GiB), xfer: 2250694 ms
 |-- Disk #0, Partition #0: Installable File System (Installable File System) Maj:Min=0:0, size: 104.9 MB
 |-- Disk #0, Partition #1: Installable File System (Installable File System) Maj:Min=0:1, size: 250.0 GB @ C:\
[main] INFO oshi.SystemInfoTest - Checking File System...
File System:
 File Descriptors: 0/0
 Local Fixed Disk (C:) (Fixed drive) [NTFS] 194.1 GiB of 232.8 GiB free (83.4%) is \\?\Volume{*******-********-********-********}\  and is mounted at C:\
[main] INFO oshi.SystemInfoTest - Checking Network interfaces...
 CD-ROM Disc (D:) (CD-ROM) [] 0 bytes of 0 bytes free (NaN%) is \\?\Volume{*******-********-********-********}\  and is mounted at D:\
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Network interfaces:
 Name: eth3 (Intel(R) 82579LM Gigabit Network Connection)
3c:d9:**:**:**:0f
########################## 194693632
   MTU: 1500, Speed: 194.7 Mbps   <---  this is the adapter of 1.GB and it says 194.7Mbps instead
##########################
   IPv4: [***********]
   IPv6: [] 
   Traffic: received 1823106 packets/190.8 MiB (0 err); transmitted 1901176 packets/160.1 MiB (0 err) 
 Name: eth4 (VMware Virtual Ethernet Adapter for VMnet1)
00:50:**:**:**:01
########################## 100000000
   MTU: 1500, Speed: 100 Mbps 
##########################
   IPv4: [***********] 
   IPv6: [**************************] 
   Traffic: received 983 packets/983 bytes (0 err); transmitted 292 packets/449.9 KiB (0 err) 
 Name: eth5 (VMware Virtual Ethernet Adapter for VMnet8)
00:50:**:**:**:08
########################## 100000000
   MTU: 1500, Speed: 100 Mbps 
##########################
   IPv4: [***********]
   IPv6: [**************************]
   Traffic: received 983 packets/983 bytes (0 err); transmitted 292 packets/545.4 KiB (0 err) 
 Name: net4 (Microsoft ISATAP Adapter #3)
00:00:00:00:00:00:00:e0
########################## 100000
   MTU: 1280, Speed: 100 Kbps 
##########################
   IPv4: [] 
   IPv6: [fe80:0:0:0:0:5efe:c0a8:be01, fe80:0:0:0:0:5efe:c0a8:c001] 
   Traffic: received ?/?; transmitted ?/? 
 Name: net5 (Microsoft ISATAP Adapter #2)
00:00:00:00:00:00:00:e0
########################## 100000
   MTU: 1280, Speed: 100 Kbps 
##########################
   IPv4: [] 
   IPv6: [**************************]
   Traffic: received ?/?; transmitted ?/? 
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
[main] INFO oshi.SystemInfoTest - Checking Network parameterss...
Network parameters:
 Host name: TRN-PC
 Domain name: TRN-PC
 DNS servers: [*********** , ***********]
 IPv4 Gateway: 
 IPv6 Gateway: 
[main] INFO oshi.SystemInfoTest - Checking Displays...
Displays:
 Display 0:
  Manuf. ID=HG, Product ID=3344, Analog, Serial=00000000, ManufDate=4/2017, EDID v1.4
  80 x 33 cm (31.5 x 13.0 in)
  Preferred Timing: Clock 319MHz, Active Pixels 5120x3840 
  Range Limits: Field Rate 30-60 Hz vertical, 30-90 Hz horizontal, Max clock: 320 MHz
  Monitor Name: HP S340c
  Serial Number: *********
[main] INFO oshi.SystemInfoTest - Checking USB Devices...
USB Devices:
 Standard Enhanced PCI to USB Host Controller ((Standard USB Host Controller))
 |-- USB Root Hub ((Standard USB Host Controller))
     |-- Generic USB Hub ((Generic USB Hub))
 Standard Enhanced PCI to USB Host Controller ((Standard USB Host Controller))
 |-- USB Root Hub ((Standard USB Host Controller))
     |-- Generic USB Hub ((Generic USB Hub))
         |-- USB Composite Device ((Standard USB Host Controller))
             |-- USB Input Device ((Standard system devices))
                 |-- HID Keyboard Device ((Standard keyboards))
             |-- USB Input Device ((Standard system devices))
                 |-- HID-compliant consumer control device (Microsoft)
                 |-- HID-compliant device ((Standard system devices))
         |-- USB Input Device ((Standard system devices))
             |-- HID-compliant mouse (Microsoft)

**Comment by dbwiddis** [comment]
OK based on the sent/received values I'm suspecting OSHI is using the 32-bit `MIB_IFROW` value rather than the 64-bit `MIB_IFROW2`.

Can you run this JNA code and see what results it gives for `majorVersion`?
```
        OSVERSIONINFO lpVersionInfo = new OSVERSIONINFO();
        // GetVersionEx() isn't accurate for Win8+ but is sufficient for
        // detecting versions 5.x and earlier
        if (!Kernel32.INSTANCE.GetVersionEx(lpVersionInfo)) {
            majorVersion = lpVersionInfo.dwMajorVersion.byteValue();
        } else {
            majorVersion = 0;
        }
```

**Comment by Davis8988** [comment]
Hi,  
I ran the code and the results:  
**majorVersion = 0.**  
  
The condition  (!Kernel32.INSTANCE.GetVersionEx(lpVersionInfo)) turns out **false**.

My operating system is Window 7 professional x64 bit

**Comment by dbwiddis** [comment]
OK.  We've confirmed this isn't about the net speed so much as my version checking code is bad.  So if the !Get() is false that means the Get() is true.  I think I switched results on this... I just reread the docs for [GetVersionEx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451(v=vs.85).aspx) and it says:

```
If the function succeeds, the return value is a nonzero value.
If the function fails, the return value is zero. 
```
Also, 

> Before calling the GetVersionEx function, set the dwOSVersionInfoSize member of the structure as appropriate to indicate which data structure is being passed to this function.

So seems like a couple bugs here.

1. Need to remove the ! on the version check boolean
2. Need to set `lpVersionInfo.dwOSVersionInfoSize` to `WinNT.OSVERSIONINFO.dwOSVersionInfoSize`

Can you test this out and see if you get the right code version?

Or, better, use the simpler version I've used elsewhere in the code:
`majorVersion = Kernel32.INSTANCE.GetVersion().byteValue();`

**Comment by dbwiddis** [comment]
While `byteValue()` works it's not obvious what's happening.  A better solution (which should be applied elsewhere) would be:
`Kernel32.INSTANCE.GetVersion().getLow().byteValue();`

**Comment by dbwiddis** [comment]
SNAPSHOT is updated with the fix

**Comment by UnAfraid** [comment]
Is this fix available at 3.4.4, because i am experiencing the exact same issue at the moment on Windows 10?
